src/common/safe_state.cpp

```diff
@@ -33,4 +43,7 @@ void hwio_safe_state(void) {
     yEnable.write(Pin::State::high);
     zEnable.write(Pin::State::high);
     e0Enable.write(Pin::State::high);
+
+    force_record_metric_zero_integer_and_disable(metric_nozzle_PWM);
+    force_record_metric_zero_integer_and_disable(metric_bed_PWM);
```

Comment:
What is the motivation for this here? Why do we need to record zero and disable those metrics in safe state?

Reply:
Because analog output pin here is changed to digital pin and its output value is set to 0. So any subsequent setting of PWM duty cycle done from hwio_....cpp has no effect on output value.

Reply:
but why do we care? We are in a safe state. We aren't going to record any metrics whatsoever

Reply:
removed